src/navigators/createMaterialBottomTabNavigator.js

```diff
@@ -18,19 +18,34 @@ class BottomNavigationView extends React.Component<Props> {
     return options.tabBarColor;
   };
 
+  _isVisible() {
+    const {
+      navigation,
+      descriptors,
+    } = this.props;
+    const {state} = navigation;
+    const route = state.routes[state.index];
+    const options = descriptors[route.key].options;
+    return options.tabBarVisible == null || options.tabBarVisible;
```

Comment:
Can you use the code I posted here instead: https://github.com/react-navigation/react-navigation-material-bottom-tabs/pull/20#discussion_r215773080 ? This won't play well if the user had changed `display` prop in `barStyle`. The code I posted will allow the user to have default visibility in the `barStyle` and override it per tab basis with `options.tabBarVisible`

Reply:
display: 'auto' isn't an option in React native apparently.

Reply:
I've tested it and it works per tab. Or am I not understanding what you mean?

Reply:
@mahyali my bad, it should be `display: null`